interop/hole-punching/hole_punching.nim

```diff
@@ -85,7 +85,9 @@ proc main() {.async.} =
     let relayId = await switch.connect(relayMA).wait(30.seconds)
     debug "Connected to relay", relayId
   except AsyncTimeoutError:
-    raise newException(CatchableError, "Connection to relay timed out")
+    raise newException(
+      CatchableError, "Connection to relay timed out: " & getCurrentExceptionMsg()
+    )
```

Comment:
Is there a difference between this and 
```nim
except AsyncTimeoutError as exc:
  error "Program execution timed out: " & exc.msg
```
Same for all occurrences of `getCurrentExceptionmsg()`

Reply:
> works in an catch-all except:

we avoid catch-all since they are undefined behaviour for Defect.

> as exc

it's better to use this style consistently as it can be problematic in nested exception handlers.